nessie/src/main/java/org/apache/iceberg/nessie/UpdateableReference.java

```diff
@@ -71,6 +73,15 @@ public Reference getReference() {
     return reference;
   }
 
+  public Reference getReferenceForApiRequest() {
```

Comment:
What's the point of returning a `Branch` object as a `Reference` if the hash's always `null`? I suspect, there's a simpler way to just use the reference name, right?

Reply:
True that we can just pass the reference name for this request. But in some cases, caller has to pass the hash (when use reference + hash is set). 

So, to avoid having a check in all the callers (whether need to pass hash or not). I am passing like this.

Reply:
>  I suspect, there's a simpler way to just use the reference name, right?

I changed to use just the reference name in the builder. But it contradicts with previous comment (handling things with less code changes)